base/irrationals.jl

```diff
@@ -60,20 +60,41 @@ AbstractFloat(x::AbstractIrrational) = Float64(x)::Float64
 Float16(x::AbstractIrrational) = Float16(Float32(x)::Float32)
 Complex{T}(x::AbstractIrrational) where {T<:Real} = Complex{T}(T(x))
 
-function _irrational_to_rational(::Type{T}, x::AbstractIrrational) where T<:Integer
-    o = precision(BigFloat)
+function _irrational_to_rational_at_current_precision(::Type{T}, x::AbstractIrrational) where {T <: Integer}
+    bx = BigFloat(x)
+    r = rationalize(T, bx, tol = 0)
+    br = BigFloat(r, precision = precision(BigFloat) + 32)
+    if eps(bx) < abs(br - bx)
```

Comment:
Don't make unrelated changes:

```suggestion
    if abs(BigFloat(r) - bx) > eps(bx)
```